iocage_lib/ioc_start.py

```diff
@@ -782,14 +782,23 @@ def start_network_interface_vnet(self, nic_defs, net_configs, jid):
 
         nic_defs = nic_defs.split(",")
         nics = list(map(lambda x: x.split(":")[0], nic_defs))
+        is_netgraph = self.get('netgraph')
```

Comment:
```console
vagrant@freebsd:/var/log % sudo iocage create -r 12.0-RELEASE -n foo vnet=on interfaces='vnet0:bridge0' vnet_default_interface='vtnet0' ip4_addr='vnet0|DHCP'
dhcp requires bpf, enabling!
foo successfully created!
vagrant@freebsd:/var/log % sudo iocage start foo
* Starting foo
  + Started OK
  + Using devfs_ruleset: 5
running command: ['ifconfig', 'epair', 'create']
running command: ['ifconfig', 'epair0a', 'name', 'vnet0.1']
running command: ['ifconfig', 'vnet0.1', 'link', '0ef84ee655c7', 'mtu', '1500']
running command: ['ifconfig', 'vnet0.1', 'description', 'associated with jail: foo']
running command: ['ifconfig', 'bridge0', 'addm', 'vnet0.1']
running command: ['ifconfig', 'epair0b', 'up']
running command: ['ifconfig', 'epair0b', 'vnet', 'ioc-foo']
running command: ['jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'mtu', '1500']
running command: ['setfib', '0', 'jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'link', '02f84ee655c7']
running command: ['ifconfig', 'vnet0.1', 'up']
  + Configuring VNET OK
  + Starting services OK
  + Executing poststart OK
  + DHCP Address: 10.0.2.17/24
vagrant@freebsd:/var/log %
```

Works on my box(tm). Just kidding. Continuing to try to replicate. It may be an issue if I leave some of those interfaces or addr parameters defaulted.

Reply:
I think you should try removing all interfaces leaving just the default one i.e em0 ?
After starting through master branch, vnet_netgraph branch started my jail with dhcp correctly as well. I removed the bridge and it faulted again

Reply:
Took it down to minimal parameters for DHCP, and still getting an address. Can you send me your `iocage get all foo`?
```console
vagrant@freebsd:/var/log % sudo iocage create -r 12.0-RELEASE -n foo vnet=on dhcp=on
dhcp requires bpf, enabling!
foo successfully created!
vagrant@freebsd:/var/log % sudo iocage start foo
* Starting foo
  + Started OK
  + Using devfs_ruleset: 5
running command: ['ifconfig', 'epair', 'create']
running command: ['ifconfig', 'epair0a', 'name', 'vnet0.4']
running command: ['ifconfig', 'vnet0.4', 'link', '0ef84ee655c7', 'mtu', '1500']
running command: ['ifconfig', 'vnet0.4', 'description', 'associated with jail: foo']
running command: ['ifconfig', 'bridge0', 'addm', 'vnet0.4']
running command: ['ifconfig', 'epair0b', 'up']
running command: ['ifconfig', 'epair0b', 'vnet', 'ioc-foo']
running command: ['jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'mtu', '1500']
running command: ['setfib', '0', 'jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'link', '02f84ee655c7']
running command: ['ifconfig', 'vnet0.4', 'up']
  + Configuring VNET OK
  + Starting services OK
  + Executing poststart OK
  + DHCP Address: 10.0.2.17/24
vagrant@freebsd:/var/log %
```

Reply:
Please try deleting the interfaces as well.
```
root@freebsdvm:~/iocage # iocage get all foo
CONFIG_VERSION:20
allow_chflags:0
allow_mlock:0
allow_mount:0
allow_mount_devfs:0
allow_mount_fusefs:0
allow_mount_nullfs:0
allow_mount_procfs:0
allow_mount_tmpfs:0
allow_mount_zfs:0
allow_quotas:0
allow_raw_sockets:0
allow_set_hostname:1
allow_socket_af:0
allow_sysvipc:0
allow_tun:0
allow_vmm:0
available:readonly
basejail:0
boot:0
bpf:1
children_max:0
cloned_release:11.2-RELEASE
comment:none
compression:lz4
compressratio:readonly
coredumpsize:off
count:1
cpuset:off
cputime:off
datasize:off
dedup:off
defaultrouter:none
defaultrouter6:none
depends:none
devfs_ruleset:4
dhcp:1
enforce_statfs:2
exec_clean:1
exec_fib:0
exec_jail_user:root
exec_poststart:/usr/bin/true
exec_poststop:/usr/bin/true
exec_prestart:/usr/bin/true
exec_prestop:/usr/bin/true
exec_start:/bin/sh /etc/rc
exec_stop:/bin/sh /etc/rc.shutdown
exec_system_jail_user:0
exec_system_user:root
exec_timeout:60
host_domainname:none
host_hostname:foo
host_hostuuid:foo
host_time:1
hostid:94f8f68b-a7b6-4bcf-a4ee-9a5e6cab968a
hostid_strict_check:0
interfaces:vnet0:bridge0
ip4:new
ip4_addr:none
ip4_saddrsel:1
ip6:new
ip6_addr:none
ip6_saddrsel:1
ip_hostname:0
jail_zfs:0
jail_zfs_dataset:iocage/jails/foo/data
jail_zfs_mountpoint:none
last_started:2019-02-12 23:27:39
login_flags:-f root
maxproc:off
memorylocked:off
memoryuse:off
mount_devfs:1
mount_fdescfs:1
mount_linprocfs:0
mount_procfs:0
mountpoint:readonly
msgqqueued:off
msgqsize:off
netgraph:0
nmsgq:off
notes:none
nsem:off
nsemop:off
nshm:off
nthr:off
openfiles:off
origin:readonly
owner:root
pcpu:off
priority:99
pseudoterminals:off
quota:none
readbps:off
readiops:off
release:11.2-RELEASE
reservation:none
resolver:/etc/resolv.conf
rlimits:off
rtsold:0
securelevel:2
shmsize:off
stacksize:off
state:down
stop_timeout:30
swapuse:off
sync_state:none
sync_target:none
sync_tgt_zpool:none
sysvmsg:new
sysvsem:new
sysvshm:new
template:0
type:jail
used:readonly
vmemoryuse:off
vnet:1
vnet0_mac:0ef84e32f09d 02f84e32f09d
vnet1_mac:none
vnet2_mac:none
vnet3_mac:none
vnet_default_interface:auto
vnet_interfaces:none
wallclock:off
writebps:off
writeiops:off
root@freebsdvm:~/iocage #
```

Reply:
@overhacked I believe there's two problems, the jail's interface is not being added to the bridge and the host interface (with auto) is also not being added to the bridge despite the commands.

```
root@iocage-12:~ # iocage start foo
* Starting foo
  + Started OK
  + Using devfs_ruleset: 6
running command: ['ifconfig', 'epair', 'create']
running command: ['ifconfig', 'epair0a', 'name', 'vnet0.5']
running command: ['ifconfig', 'vnet0.5', 'link', '06f84e000000', 'mtu', '1500']
running command: ['ifconfig', 'vnet0.5', 'description', 'associated with jail: foo']
running command: ['ifconfig', 'bridge0', 'addm', 'vnet0.5']
running command: ['ifconfig', 'epair0b', 'up']
running command: ['ifconfig', 'epair0b', 'vnet', 'ioc-foo']
running command: ['jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'mtu', '1500']
running command: ['setfib', '0', 'jexec', 'ioc-foo', 'ifconfig', 'epair0b', 'link', '0af84e000000']
running command: ['ifconfig', 'vnet0.5', 'up']
  + Configuring VNET OK
  + Starting services OK
  + Executing poststart OK
  + Acquiring DHCP address: FAILED, address received: 0.0.0.0/8

Stopped foo due to DHCP failure
root@iocage-12:~ # ifconfig
vtnet0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=6c07bb<RXCSUM,TXCSUM,VLAN_MTU,VLAN_HWTAGGING,JUMBO_MTU,VLAN_HWCSUM,TSO4,TSO6,LRO,VLAN_HWTSO,LINKSTATE,RXCSUM_IPV6,TXCSUM_IPV6>
	ether 9a:1b:a4:2e:4e:43
	inet 192.168.1.18 netmask 0xffffff00 broadcast 255.255.255.0 
	media: Ethernet 10Gbase-T <full-duplex>
	status: active
	nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
	options=680003<RXCSUM,TXCSUM,LINKSTATE,RXCSUM_IPV6,TXCSUM_IPV6>
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x2 
	inet 127.0.0.1 netmask 0xff000000 
	groups: lo 
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
ngeth0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=28<VLAN_MTU,JUMBO_MTU>
	ether 00:00:00:00:00:00
	inet 192.168.1.19 netmask 0xffffff00 broadcast 192.168.1.255 
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
	nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
vnet0.4: flags=8842<BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	description: associated with jail: foo
	options=28<VLAN_MTU,JUMBO_MTU>
	ether 06:f8:4e:00:00:00
	media: Ethernet autoselect (1000baseT <full-duplex>)
	status: active
	nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
bridge0: flags=8802<BROADCAST,SIMPLEX,MULTICAST> metric 0 mtu 1500
	ether 02:91:ce:37:6d:00
	id 00:00:00:00:00:00 priority 32768 hellotime 2 fwddelay 15
	maxage 20 holdcnt 6 proto rstp maxaddr 2000 timeout 1200
	root id 00:00:00:00:00:00 priority 0 ifcost 0 port 0
	member: ngeth0 flags=143<LEARNING,DISCOVER,AUTOEDGE,AUTOPTP>
	        ifmaxaddr 0 port 3 priority 128 path cost 20000
	groups: bridge 
	nd6 options=9<PERFORMNUD,IFDISABLED>
```